docs/DeveloperGuide.md

```diff
@@ -166,8 +166,143 @@ Below shows the sequence diagram for ```ListAppointmentCommand``` class
 
 
 #### 2.2.4 Parser module
-**TODO**
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+This section describes the implementation of Parser class, as well as the design considerations and rational behind the 
+current implementation.
+
+The main purpose of the Parser class is as below.
+1.  To interpret user inputs so that the correct command can be executed 
+2.  Functions as the first line to sanitize user input
+
+As such, the Parser class only has one publicly callable method and that returns the command object for execution. 
+To assist the public method, Parser class has multiple private helper methods, as well as access to the Exception handler
+to ensure that the user input is formatted correctly. This way, purpose 1 and 2 is satisfied. 
+
+##### 2.2.4.1 Object creation and steps in input interpretation
+1.  The `Parser` object is first created in `Duke` class and subsequently used until program termination.
+2.  User input is received and handed over to `Parser` object for interpretation.
+3. In the `Parser` object, the type of command is first determined via helper method `getCommand(userInput)`.
+    + Example: `addp \age 23 \name Justin`, the `addp` command will be determined.
+4. The remaining fields will be recorded in a hashMap through either `fillPatientFields(userInput)` or `fillAppointmentFields(userInput)`
+depending on the command type in Step 3.
+    + The type of category of the command can be determined based on the last alphabet of the first word. 
+    + The command `addp` has the last alphabet is `p`, `fillPatientFields()` will be called. 
+    Likewise, if it ise `adda`, it will be `fillAppointmentFields()`.
+    + Example: `addp \name Sam \age 18`, the method `fillPatientFields(userInput)` will be called. The hashMap will contain
+    age -> 23, name -> Sam. 
+5. At the end of the execution, a reference to the command object will be returned. 
+
+>![](images/SD_parser/capture2.JPG)
+
+>![](images/SD_parser/capture.JPG)
+
+Sequence Diagram when `parseCommand(userInput)` is initially called
+![](images/SD_parser/Slide1.JPG)
```

Comment:
inconsistent spacing at start of each alt for eg: [Edit Patient] has one more spacing at front than [Edit Appointment]